manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -112,13 +120,19 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         ProcessorDefinition definition = new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, requestedAction, resolvedAction);
         newProcessor.setDefinition(definitionToJsonNode(definition));
 
-        processorDAO.persist(newProcessor);
+        Optional<ConnectorEntity> persist = persistProcessorCreateConnector(newProcessor, actionProvider, resolvedAction);
 
-        connectorService.createConnectorIfNeeded(resolvedAction, newProcessor, actionProvider);
+        persist.ifPresent(c -> eventBus.requestAndForget(CONNECTOR_CREATED_EVENT, c));
 
         return newProcessor;
     }
 
+    @Transactional
```

Comment:
I don't think the `@Transactional` annotation works on private methods. Is there any reason why you removed it from the public `createProcessor` method?

Reply:
Good catch, it was the ConnectorService opening a transaction because we changed from `MANDATORY` to `REQUIRES_NEW` in the previous PR. I'm setting Connector's back to `MANDATORY` with a comment